### PR TITLE
feat: add pipenv completion spec

### PR DESCRIPTION
**Pull request by ly0708**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

1. Adding a completion spec for pipenv
2. Updated pip.ts to export the packageList and use it at pipenv

**What is the current behavior? (You can also link to an open issue here)**

Currently there is no completion for pipenv

**What is the new behavior (if this is a feature change)?**

This adds autocomplete for pipenv, and other files can use the packageList in pip.ts by importing the variable

**Additional info:**

### CODE REVIEW

File: dev/pip.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ const listPackages: Fig.Generator = {
   },
 };
 
+export { packageList }; // To use it - import { packageList } from "./pip"
 const packageList: Array<Fig.Suggestion> = [
```
Comment (fedeci) [inline dev/pip.ts:17]:
```suggestion
// Accessible through ES imports - e.g. import { packageList } from "./pip"
export const packageList: Array<Fig.Suggestion> = [
```

File: dev/pipenv.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      description: "Display manpage",
+    },
+
+    {
+      name: "--support",
+      description: "View diagnostic information for use in Github issues",
+    },
+
+    {
+      name: ["--site-packages", "--no-site-packages"],
```
Comment (fedeci) [inline dev/pipenv.ts:130]: Aren't these two different commands each one `exclusiveOn` the other?

File: dev/pipenv.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    {
+      name: ["-v", "--verbose"],
+      description: "Verbose mode",
+    },
+
+    {
+      name: "--pypi-mirror",
+      description: "Specify a PyPi mirror",
+      args: {
+        name: "TEXT",
```
Comment (fedeci) [inline dev/pipenv.ts:168]:
```suggestion
        name: "mirror",
```

File: dev/pipenv.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
+import { packageList } from "./pip";
```
Comment (fedeci) [inline dev/pipenv.ts:1]: There are some options missing in the spec, if you do not intend to add those in this PR I would add this message for future contributors
```suggestion
// https://pipenv.pypa.io/en/stable/cli/
// TODO: complete spec
import { packageList } from "./pip";
```

**Review by fedeci — COMMENTED** [review]
Could you please remove the update `package-lock.json`? We are using npm@6

**Comment by ly0708** [comment]
package-lock.json issue is still not resolve

**Review by fedeci — COMMENTED** [review]
I still have to give a look at the updated spec itself, can you please run the following git command in the root of the repo? It will rebase just one file to the specified commit.
`git checkout 0433c25 -- ./package-lock.json`

**Comment by ly0708** [comment]
I am getting 
`fatal: invalid reference: 0433c25`

**Comment by fedeci** [comment]
Probably it want the entire hash.
git checkout 0433c2561ab69bf824fb845da9df5f943bd54ff0 -- ./package-lock.json

**Comment by ly0708** [comment]
@fedeci  Sorry. I am going to need help with this. Would you mind hop on a call with me via discord? If so, what is your time zone? I am in Central Time (CDT). And I am in school, so either late at night (CDT) or weekends will be the best for me.

**Comment by ly0708** [comment]
hmm. now I am getting. 
`fatal: reference is not a tree: 0433c2561ab69bf824fb845da9df5f943bd54ff0`

**Review by QuiiBz — APPROVED** [review]
LGTM! I reset back the `package-lock.json` manually. Thank you so much for this PR 🎉

**Comment by fedeci** [comment]
@ly0708 FYI it was not working because I forgot to tell you to pull and rebase on top of the current main branch. Sorry!

**Pull request merged**